Automattic/jetpack
# VideoPress: Add confirmation before leaving video details edit component when there are unsaved changes

Fixes #26917

#### Changes proposed in this Pull Request:

* Adds confirmation when leaving the video details edit form

Note: There are two ways of leaving the form: leaving the page entirely and leaving only the `EditVideoDetails` component.
When leaving the page entirely we need to use the `beforeunload` event listener as used [here](https://github.com/Automattic/jetpack/pull/26912/files#diff-26d9236c185def582b6cb006647b7812fba8bd9f8aec6b4f7b6d0c426dcff2b6R19), but when navigating between internal routes the page is not unloaded, so we need to use functionality from the router.
Version 6 of `react-router-dom` [currently does not support this](https://reactrouter.com/en/6.4.1/upgrading/v5#prompt-is-not-currently-supported), so a downgrade is necessary until then.

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Go to the VideoPress dashboard
* Click on `Edit video details`
* Do not change anything. Check that no message is shown when trying to navigate away from the form
* Change something and try to navigate away from the form. A message should appear prompting the user to stay in the page

When leaving page:
![2022-10-18_17-21-19](https://user-images.githubusercontent.com/8486249/196539366-67dd3855-d249-46e1-90cc-f2c57535f749.png)

When clicking on the VideoPress link to go back to the videos component:
![2022-10-18_17-36-31](https://user-images.githubusercontent.com/8486249/196539742-0b943f32-b416-4060-9d7a-6a72f21d7aa3.png)
